homeassistant/components/lupusec/__init__.py

```diff
@@ -39,44 +41,77 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-LUPUSEC_PLATFORMS = [
+LUPUSEC_PLATFORMS: list[Platform] = [
     Platform.ALARM_CONTROL_PANEL,
     Platform.BINARY_SENSOR,
     Platform.SWITCH,
 ]
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up Lupusec component."""
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the lupusec integration."""
+
+    if DOMAIN not in config:
+        return True
+
     conf = config[DOMAIN]
-    username = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
-    ip_address = conf[CONF_IP_ADDRESS]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+    host = conf.get(CONF_IP_ADDRESS)
     name = conf.get(CONF_NAME)
 
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_USERNAME: username,
+                CONF_PASSWORD: password,
+                CONF_HOST: host,
+                CONF_FRIENDLY_NAME: name,
```

Comment:
All values are already passed as required in the config schema, might as well just pass conf as data in the `async_init`

Reply:
Not possible: changed CONF_NAME in the config_flow to CONF_FRIENDLY_NAME, 
name, however is legacy in yaml.

Reply:
Sooo? That doesn't mean we can't exclude it from the config flow

Reply:
Done.